SU2_CFD/src/solvers/CTransAFTSolver.cpp

```diff
@@ -0,0 +1,598 @@
+    const su2double Laminar_Viscosity_i = flowNodes->GetLaminarViscosity(iPoint);
+    const su2double Density_i = flowNodes->GetDensity(iPoint);
+    const su2double c_1 = 100.0;
+    const su2double c_2 = 0.06;
+    const su2double c_3 = 50.0;
+    su2double Temp3 = flowNodes->GetVelocity(iPoint, 0) * nodes->GetAuxVarGradient(iPoint, 0, 0);
+    su2double Temp4 = flowNodes->GetVelocity(iPoint, 1) * nodes->GetAuxVarGradient(iPoint, 0, 1);
+
+    if(nDim == 2) {
+      nodes->SetAuxVar(iPoint, 1, Temp3 + Temp4);
+    }
+    else {
+      nodes->SetAuxVar(iPoint, 1, Temp3 + Temp4 + flowNodes->GetVelocity(iPoint, 2) * nodes->GetAuxVarGradient(iPoint, 0, 2));
+    }
+
+    su2double VorticityMag = GeometryToolbox::Norm(3, flowNodes->GetVorticity(iPoint));
+    su2double HLGradTerm = 0.0;
+    HLGradTerm = nodes->GetAuxVarGradient(iPoint, 1, 0) * nodes->GetAuxVarGradient(iPoint, 0, 0) + nodes->GetAuxVarGradient(iPoint, 1, 1) * nodes->GetAuxVarGradient(iPoint, 0, 1);
+
+    if(nDim == 3) {
+      HLGradTerm += nodes->GetAuxVarGradient(iPoint, 1, 2) * nodes->GetAuxVarGradient(iPoint, 0, 2);
+    }
+
+    /*--- Cal H12, Hk, dNdRet, Ret0 ---*/
+    su2double HL = min( max(dist_i * dist_i * Density_i / Laminar_Viscosity_i  * HLGradTerm, -0.25), 200.0);
+    const su2double H12 = TransCorrelations.H12_Correlations(HL);
+    const su2double dNdRet = TransCorrelations.dNdRet_Correlations(H12);
+    const su2double Ret0 = TransCorrelations.Ret0_Correlations(H12);
+    const su2double D_H12 = TransCorrelations.D_Correlations(H12);
+    const su2double l_H12 = TransCorrelations.l_Correlations(H12);
+    const su2double m_H12 = TransCorrelations.m_Correlations(H12, l_H12);
+    const su2double kv = TransCorrelations.kv_Correlations(H12);
+    const su2double Rev = Density_i * dist_i * dist_i * StrainMag_i / (Laminar_Viscosity_i + Eddy_Viscosity_i);
+    const su2double Rev0 = kv * Ret0;
+
+    /*--- production term of the amplification factor ---*/
+    const su2double F_growth = max(D_H12 * (1.0 + m_H12) / 2.0 * l_H12, 0.0);
+    const su2double F_crit = (Rev < Rev0) ? 0.0 : 1.0;
+    const su2double PAF = Density_i * VorticityMag * F_crit * F_growth * dNdRet;
+
+    /*--- production term of the intermittency(Gamma) ---*/
+    const su2double RT = Eddy_Viscosity_i / Laminar_Viscosity_i;
+    const su2double F_onset1 = AF / Critical_N_Factor;
+    const su2double F_onset2 = min(F_onset1, 2.0);
+    const su2double F_onset3 = max(1.0 - pow(RT / 3.5, 3), 0.0);
+    const su2double F_onset = max(F_onset2 - F_onset3, 0.0);
+    const su2double F_turb = exp(-pow(RT / 2.0,4));
+    const su2double Pg = c_1 * Density_i * StrainMag_i * F_onset * (1.0 - exp(lnIntermittency));
+
+    /*-- destruction term of Intermeittency(Gamma) --*/
+    const su2double Dg = c_2 * Density_i * VorticityMag * F_turb * (c_3 * exp(lnIntermittency) - 1.0);
+    nodes -> SetAFT_Wonder_Func(iPoint, HL, H12, dNdRet, Ret0, D_H12, l_H12, m_H12, kv, Rev, Rev0, F_growth, F_crit, PAF, Pg, Dg);
```

Comment:
This is a lot of memory and these variables are only used for post processing (output) right?
If so these can be computed in the output class when setting the output values.
That saves doing this work every iteration and storing all these variables.

Reply:
Thank you for your advise.
Current class, this function is for debugging. Later, I'll change to compute in the output class.